corehq/apps/custom_data_fields/models.py

```diff
@@ -50,9 +50,10 @@ class CustomDataFieldsDefinition(Document):
     domain = StringProperty()
     fields = SchemaListProperty(CustomDataField)
 
-    def get_fields(self, required_only=False):
+    def get_fields(self, required_only=False, include_system=True):
         def _is_match(field):
-            if required_only and not field.is_required:
+            if (required_only and not field.is_required
+                    or not include_system and is_system_key(field.slug)):
```

Comment:
(not blocking) although this works using the order of evaluation of ands and ors, i usually prefer to make it explicit with parentheses so it's easier to read